500-CNAB-dependencies.md

```diff
@@ -0,0 +1,100 @@
+a database as a service on a cloud platform, and then use that bundle as a 
+dependency.
+
+This specification defines [dependencies metadata](#dependencies-metadata) in the 
+bundle.json for specifying dependencies but does not dictate that the metadata is
+specifically used at a particular time, or how the dependency graph is resolved. 
+Each tool may choose to use the information differently, for example [porter](https://porter.sh) resolves dependencies when the bundle is built, but another tool could 
+use this information at runtime.
+
+There are two cases for how a bundle may need to depend upon another bundle:
+
+1. [Depend on a named bundle](#depend-on-a-named-bundle)
+1. [Depend on a bundle that provides a capability](#depend-on-a-bundle-that-provides-a-capability)
+
+## Depend on a named bundle
+
+The bundle depends on a specific named bundle that is known in advance.
+
+```json
+{
+    "name": "wordpress",
+    "dependencies": [
+        {
+            "requires":{
+                "bundle": "azure/mysql",
+                "version": {
+                    "range": "5.7.x",
+                    "prereleases": "true"
+                }
+            }
+        }
+    ]
+}
+```
+
+## Depend on a bundle that provides a capability
+
+This section is a placeholder and will be completed in a follow-up pull request.
+
+## Dependencies Metadata
+
+This specification introduces a `dependencies` array in the bundle.json
+that defines metadata necessary to specify a dependency.
+
+### requires
+
+The `requires` object defines the criteria for the dependent bundle:
+
+* `bundle`: A reference to a bundle in the format REGISTRY/NAME.
+* `version`: A set of criteria applied to the bundle version when selecting an
+    acceptable version of the bundle.
+    * `ranges`: An array of allowed version ranges. 
+    
+        Versions are specified using semver, with or without a leading "v". 
+        An `x` in the place of the minor or patch place can be used to specify 
+        a wildcard. Ranges can be specified by separating the two versions with 
+        a dash, the dash must be surrounded by spaces in order to disambiguate 
+        from prerelease tags.
+
+        Below are some example ranges:
+        * `1.2.3` - Requires an exact version.
+        * `1.2.x` - Restricts the major.minor to `1.2` and allows any patch.
+        * `1.x` - Restricts the major to `1` and allows any minor.
+        * `1.5.x - 3` - Allows a range of versions from `1.5` up to but not including 4.
+    * `prereleases`: A boolean field, defaults to `false`, which specifies if
+        prerelease versions of the bundle are allowed.
+
+# Q & A
+
+This isn't meant to be in the final spec but gives people a place to comment
+on anticipated questions.
+
+* Why didn't use you `~>` and other cute symbols?
+    * Because they have no global meaning across languages and frameworks and are hard to understand at a glance.
```

Comment:
I don't find that answer convincing: (a) The meaning is relatively well established across most major package managers, and (b) as long as you define the meanings, it doesn't really matter what others do.

An analogous argument would be "why doesn't your programming language have a string concat operator"; "because other languages use different characters for string concat operators."

Reply:
Just to be clear, I'm not suggesting that we MUST have those operators. I'm merely saying that if that is the only reason, I don't find it to be a good one.

The more I think about it, the more I think this is actually not something you even have to address in the spec. If the .x notation works for what you need, just don't bother distracting people with this discussion.

Reply:
Per the comment under the heading, these Q&A's aren't going into the spec. They are just a place for me to proactively ask and answer expected questions about what I did in this proposal.